File: baselibrary/src/main/java/com/minicart/android/baselibrary/mvp/IPresenter.java
package com.minicart.android.baselibrary.mvp;

/**
 * @类名：IPresenter
 * @描述：
 * @创建人：54506
 * @创建时间：2017/1/13 11:04
 * @版本：
 */

public interface IPresenter<V extends IView> {

    String SORT_TYPE = "01";
    String PAGE_SIZE = "2000";
    String CURRENT_PAGE = "1";

    void injectView(V view);

    void onAttachView();

    void onStart();

    void onRestart();

    void onResume();

    void onPause();

    void onStop();

    void onDestroy();

}

File: baselibrary-kotlin/src/main/java/com/minicart/android/baselibrary/mvp/IPresenter.kt
package com.minicart.android.baselibrary.mvp

/**
 * @类名：IPresenter
 * *
 * @描述：
 * *
 * @创建人：54506
 * *
 * @创建时间：2017/1/13 11:04
 * *
 * @版本：
 */

interface IPresenter<in V> {

    fun injectView(view: V)

    fun onAttachView()

    fun onStart()

    fun onRestart()

    fun onResume()

    fun onPause()

    fun onStop()

    fun onDestroy()

    companion object {

        val SORT_TYPE = "01"
        val PAGE_SIZE = "2000"
        val CURRENT_PAGE = "1"
    }

}
